From 56f6eba87c1cef230c390116b0c35a907bb97eff Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Tue, 28 Oct 2025 09:14:55 +0100
Subject: [PATCH] bugfix: remove spurious GPIO15 interrupt

---
 src/main.c   | 2 +-
 src/system.c | 8 ++------
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/main.c b/src/main.c
index b9f1b04..5424d79 100644
--- a/src/main.c
+++ b/src/main.c
@@ -136,7 +136,7 @@ int main(void)
 
   watchdog_init(IWDG_PRESCALER_16);
 
-  gpio_init_ncs();
+  //gpio_init_ncs();
 
   for(;;) {
     handle_data();
diff --git a/src/system.c b/src/system.c
index 23dcf32..f7907a7 100644
--- a/src/system.c
+++ b/src/system.c
@@ -369,12 +369,7 @@ void dma_load(bool const swap_tx_buf)
 
 void EXTI15_10_IRQHandler(void)
 {
-  /* PA15 = nCS */
-  if (HAL_GPIO_ReadPin(GPIOA, GPIO_PIN_15) == GPIO_PIN_SET && LL_EXTI_ReadFlag_0_31(LL_EXTI_LINE_15)) {
-    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_15);
-  } else {
-    gpio_handle_irq();
-  }
+  gpio_handle_irq();
 }
 
 void HAL_SPI_TxRxCpltCallback(SPI_HandleTypeDef *hspi)
@@ -408,6 +403,7 @@ void HAL_SPI_ErrorCallback(SPI_HandleTypeDef *hspi)
 
   /* Preload buffers for next communication. */
   dma_load(false);
+  clean_dma_buffer();
   set_nirq_high();
 }
 
-- 
2.51.0

